src/sqlacodegen/generators.py

```diff
@@ -1358,17 +1369,23 @@ def __init__(
             base_class_name=base_class_name,
         )
 
+    @property
+    def mapped_columns_supported(self) -> bool:
+        return False
+
     def generate_base(self) -> None:
         self.base = Base(
             literal_imports=[],
             declarations=[],
-            metadata_ref="",
+            metadata_ref="metadata",
+            table_metadata_declaration="metadata = MetaData()",
         )
 
     def collect_imports(self, models: Iterable[Model]) -> None:
         super(DeclarativeGenerator, self).collect_imports(models)
+        if any(isinstance(model, Model) for model in models):
+            self.add_literal_import("sqlalchemy", "MetaData")
```

Comment:
Separate unrelated control blocks with a blank line for readability.
```suggestion
            self.add_literal_import("sqlalchemy", "MetaData")

```

Reply:
But also, why are we unconditionally importing MetaData, particularly in the case of SQLModels which don't even use it directly?

Reply:
Part of the broken codegen was a Table def being generated for SQL views without metadata. The thing is, imports are being collected / generated before the models/tables are being generated, so trying to resolve that decision is going to introduce more complexity, and if the price was having an unused `MetaData`  definition at the top of the generated python module, I assumed it was a price worth paying. Let me know if you feel this is a blocker for merging this PR.

Reply:
Was it always your intention to generate a table represenation for views when using the `sqlmodel` generator

Reply:
@agronholm so the decision should be to skip views for sqlmodel?

Reply:
@agronholm please share your decision

Reply:
If views cannot be represented as SQLModel models, then they should be skipped, but do check first if that is the case.

Reply:
@agronholm it's not that they cannot be represented, you can technically, create a model matching a view and query it. However it will be a read-only model, which is not something that technically exists in SQLModel. Views do not really fit ORM CRUD operations paradigm. My suggestion would be to drop view generation for SQLModel. If the user still want view bindings, he can just generate tables alongside the sqlmodels.

That's my call, let me know if you agree, and I'll get to it.

Reply:
Yeah, skip views then. But make sure there's at least a warning generated when this is done, so users won't be left thinking this is a bug.